validation/longhorn/longhorn_test.go

```diff
@@ -0,0 +1,314 @@
+	require.NoError(l.T(), err)
+
+	expectedSettings := map[string]string{
+		"default-data-path":                    "/var/lib/longhorn-custom",
+		"default-replica-count":                "2",
+		"storage-over-provisioning-percentage": "150",
+		"create-default-disk-labeled-nodes":    "true",
+	}
+
+	for setting, expectedValue := range expectedSettings {
+		getSettingCommand := []string{"kubectl", "-n", longhornNamespace, "get", "settings.longhorn.io", setting, `-o=jsonpath='{.value}'`}
+		settingValue, err := kubectl.Command(l.client, nil, l.cluster.ID, getSettingCommand, "")
+		require.NoError(l.T(), err)
+		// The output extracted from kubectl has single quotes and a newline on the end.
+		require.Equal(l.T(), fmt.Sprintf("'%s'\n", expectedValue), settingValue)
+	}
+
+	// Use the "longhorn-static" storage class so we get the expected number of replicas.
+	// Using the "longhorn" storage class will always result in 3 volume replicas.
+	l.T().Logf("Create nginx deployment with %s PVC on default namespace", longhornStaticDefaultStorageClass)
+	nginxResponse := cloudprovider.CreatePVCWorkload(l.T(), l.client, l.cluster.ID, longhornStaticDefaultStorageClass)
+
+	// Fetch the name of the created volume from the nginx response.
+	nginxSpec := &appv1.DeploymentSpec{}
+	err = steveV1.ConvertToK8sType(nginxResponse.Spec, nginxSpec)
+	require.NoError(l.T(), err)
+	require.NotEmpty(l.T(), nginxSpec.Template.Spec.Volumes[0])
+
+	// Even though the Longhorn default for number of replicas is 2, Rancher enforces its own default of 3.
+	volumeName := nginxSpec.Template.Spec.Volumes[0].Name
+	checkReplicasCommand := []string{"kubectl", "-n", longhornNamespace, "get", "volumes.longhorn.io", volumeName, `-o=jsonpath="{.spec.numberOfReplicas}"`}
+	settingValue, err := kubectl.Command(l.client, nil, l.cluster.ID, checkReplicasCommand, "")
+	require.NoError(l.T(), err)
+	require.Equal(l.T(), fmt.Sprintf("\"%v\"\n", expectedSettings["default-replica-count"]), settingValue)
+
+	// Create a new namespace and a debug pod within it to check the host filesystem for the custom Longhorn data directory.
+	// We do this in a separate namespace to ease cleanup.
+	l.T().Logf("Create namespace [%v] to check node filesystem with debugger pod", debugNamespace)
+	createdNamespace, err := namespaceActions.CreateNamespace(l.client, debugNamespace, "{}", nil, nil, l.project)
+	require.NoError(l.T(), err)
+
+	steveClient, err := l.client.Steve.ProxyDownstream(l.cluster.ID)
+	require.NoError(l.T(), err)
+
+	l.session.RegisterCleanupFunc(func() error {
+		return steveClient.SteveType(namespaceActions.NamespaceSteveType).Delete(createdNamespace)
+	})
+
+	checkDataPathCommand := []string{"kubectl", "debug", "node/" + workerName, "-n", debugNamespace, "--profile=general", "--image=busybox", "--", "/bin/sh", "-c", "test -d /host/var/lib/longhorn-custom/replicas && test -f /host/var/lib/longhorn-custom/longhorn-disk.cfg"}
```

Comment:
is there a reason you use kubectl instead of rancher api for stuff like this? Or in other words, is there not a way to check this via rancher instead of kubectl?